packages/core/src/binary-data/file-system.manager.ts

```diff
@@ -165,10 +189,39 @@ export class FileSystemManager implements BinaryData.Manager {
 	 * The legacy ID format `{executionId}{uuid}` for `filesystem` mode is
 	 * no longer used on write, only when reading old stored execution data.
 	 */
-	private toFileId(workflowId: string, executionId: string) {
-		if (!executionId) executionId = 'temp'; // missing only in edge case, see PR #7244
+	private toFileId(location: BinaryData.FileLocation) {
+		switch (location.type) {
+			case 'execution': {
+				const executionId = location.executionId || 'temp'; // missing only in edge case, see PR #7244
+				return `workflows/${location.workflowId}/executions/${executionId}/binary_data/${uuid()}`;
+			}
+			case 'chat-hub-message-attachment':
+				return `chat-hub/sessions/${location.sessionId}/messages/${location.messageId}/binary_data/${uuid()}`;
+		}
+	}
+
+	private parseFileId(fileId: string): BinaryData.FileLocation | null {
+		const executionMatch = fileId.match(EXECUTION_PATH_MATCHER);
+
+		if (executionMatch) {
+			return {
+				type: 'execution',
+				workflowId: executionMatch[1],
+				executionId: executionMatch[2],
+			};
+		}
+
+		const chatHubMatch = fileId.match(CHAT_HUB_ATTACHMENT_PATH_MATCHER);
+
+		if (chatHubMatch) {
+			return {
+				type: 'chat-hub-message-attachment',
+				sessionId: chatHubMatch[1],
+				messageId: chatHubMatch[2],
+			};
+		}
 
-		return `workflows/${workflowId}/executions/${executionId}/binary_data/${uuid()}`;
+		return null;
```

Comment:
I think this should not happen, right? Should we report this using the errorReporter?

Reply:
added warning via errorReporter